apps/internal/oauth/resolvers.go

```diff
@@ -120,8 +120,10 @@ func (m *authorityEndpoint) addCachedEndpoints(authorityInfo authority.Info, use
 }
 
 func (m *authorityEndpoint) openIDConfigurationEndpoint(ctx context.Context, authorityInfo authority.Info, userPrincipalName string) (string, error) {
-	if authorityInfo.Tenant == "adfs" {
+	if authorityInfo.AuthorityType == authority.ADFS {
```

Comment:
it seemed to me that using the `AuthorityType` is probably the prefered way to determine the flow here, because previously we were looking for `adfs` in `tenant` field, which felt kinda on the hacky side

another approach we could take is to make both adfs and dSTS fall through to the last line of this method and simply append the well known path, but I am now seeing adfs does not have `v2.0/.well-known` for some reason - do you happen to know why? can someone validate if the v2.0/ is supported?